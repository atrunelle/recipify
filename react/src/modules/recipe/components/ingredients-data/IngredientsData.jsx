import React, { Component } from 'react';
import './IngredientsData.css';

import { Card, CardContent, Typography, Grid } from '@material-ui/core';
import NutrientDetails from '../nutritient-details/NutrientDetails';

class IngredientsData extends Component {
  render() {
    if (!this.props.totalNutrients || !this.props.totalNutrients.length) return false;

    const nutrientsDetails = this.props.totalNutrients.map((nutrient, index) => 
        <div className="u-text-center" key={index}>
          <NutrientDetails key={index} nutrient={nutrient}/>
        </div>
      );

    return(
      <Card>
        <CardContent>
          <Typography
            gutterBottom 
            variant="headline"
            component="h2">
            Total nutrients
          </Typography>          
          <Typography
            gutterBottom 
            component="h3">
            Calories: { this.props.totalCalories }cal
          </Typography>
          <Typography
            gutterBottom 
            component="h3">
            Total Weight: { this.props.totalWeight }gr
          </Typography>
          <Grid
            container 
            alignItems="flex-end"
            justify="space-between">
            <header>
              <Typography
                gutterBottom
                paragraph
                component="p">
                Total Nutrients
              </Typography>
              <Typography
                gutterBottom
                paragraph
                component="p">
                  Total Daily percentage
              </Typography>
            </header>
            {nutrientsDetails}
          </Grid>
        </CardContent>
      </Card>
    );
  }
}

export default IngredientsData;