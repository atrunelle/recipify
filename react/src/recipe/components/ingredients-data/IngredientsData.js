import React, { Component } from 'react';
// import './IngredientsData.css';


class IngredientsData extends Component {

    render() {
        if (!this.props.ingredients.healthLabels) return (<div></div>);
        const healthLabels = this.props.ingredients.healthLabels.reduce((acc, h) => {
            return `${acc} <br> ${h}`;
        })
        return(
            <div className="ingredients">
                <h2>{this.props.title}</h2>
                <div>{this.props.ingredients.calories}</div>
                <div>{this.props.ingredients.totalWeight}</div>
                <div>{this.props.ingredients.dietLabels}</div>
                <div>{healthLabels}</div>
                {/* {this.props.ingredients.totalDaily} */}
                {/* {this.props.ingredients.totalNutrients} */}
                <div>{this.props.ingredients.cautions}</div>
                <div>{this.props.ingredients.glycemicIndex}</div>
                {/* {this.props.ingredients.ingredients} */}
            </div>
        );
    }
}

export default IngredientsData;