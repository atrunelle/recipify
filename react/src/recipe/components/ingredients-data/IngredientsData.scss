.results {
    &__title {
        font-size: 14px;
    }
}