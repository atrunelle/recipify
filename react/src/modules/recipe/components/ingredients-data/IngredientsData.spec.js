import React from 'react';
import { shallow } from 'enzyme';
import IngredientsData from './IngredientsData';

describe('Component: IngredientsData', () => {
  it('should not render if totalNutrients props is not passed', () => {
    const component = shallow(<IngredientsData />);
    expect(component.html()).toBeNull();
  });

  it('should match snapshot', () => {
    const mockTotalNutrients = [{
      totalQuantity: [{
        label: 'Fat',
        quantity: 10,
        unit: 'gr',
      }],
      totalDaily: [{
        label: 'Fat',
        quantity: 50,
        unit: '%',
      }],
    }];
    const mockTotalCalories = 200
    const mockTotalWeight = 100;


    const component = shallow(
      <IngredientsData
      totalNutrients={mockTotalNutrients}
      totalCalories={mockTotalCalories}
      totalWeight={mockTotalWeight}/>);
    expect(component.html()).toMatchSnapshot();
  });
});