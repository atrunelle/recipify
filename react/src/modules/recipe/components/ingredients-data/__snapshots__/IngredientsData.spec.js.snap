// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IngredientsData should match snapshot 1`] = `"<div class=\\"MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1\\"><div class=\\"MuiCardContent-root-29\\"><h2 class=\\"MuiTypography-root-30 MuiTypography-headline-35 MuiTypography-gutterBottom-47\\">Total nutrients</h2><h3 class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-47\\">Calories: 200cal</h3><h3 class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-47\\">Total Weight: 100gr</h3><div class=\\"MuiGrid-container-54 MuiGrid-align-items-xs-flex-end-64 MuiGrid-justify-xs-space-between-73\\"><header><p class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-47 MuiTypography-paragraph-48\\">Total Nutrients</p><p class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-47 MuiTypography-paragraph-48\\">Total Daily percentage</p></header><div class=\\"u-text-center\\"><div><h3 class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-47\\"></h3><p class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-47 MuiTypography-paragraph-48\\">NaN </p><p class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-47 MuiTypography-paragraph-48\\">NaN </p></div></div></div></div></div>"`;
